### ISSUE TITLE
Consider make quote for identifier optional in the language

### ISSUE THREAD

**Issue by edsDev** [LingDong-/wenyan-lang#236]
This is totally able to be parsed and will make the language more natural. As for readability, identifier will be colorized anyway.

**Comment by LingDong-** [comment]
Hi, thank you for the proposal.
As much as I would love to make quotes optional, they are in fact required for parsing. For my detailed explanation, please see my comment at:
https://github.com/LingDong-/wenyan-lang/issues/109#issuecomment-567109132
Thanks!

**Comment by edsDev** [comment]
Sure ambiguity would be introduced sometimes, yet for most of the time, program should still be parsed without explicitly quoted identifiers. Making quote optional also means we could quote them when necessary. This might introduce quite a lot of works though.
By the way, a strong endorsement for language spec :)

**Comment by AshyAzure** [comment]
@edsDev 
When the parser meet `「` it make the int(identifier) flag true, and when it meet `」` it push the identifier into tokens array.
If `「」` is omitted, it is very difficult for the traditional while-tokenizer to tokenize these.
Maybe considering another parsing mechanism may help.

**Comment by edsDev** [comment]
Yeah, tokenization process will be context sensitive. So implementing this will need a lot of labors.

**Comment by lymslive** [comment]
吾有一计，请君思之。

当以单字表示变量名时，省略引号，如
```
名之曰甲。
名之曰乙。
```

当以多字符表示变量名时，不可省略引号，如
```
名之曰‘甲物’。
```

这是有可能实现的，虽然也复杂，但还不至于不可行。
就像许多命令行工具，短选项之后允许不加空格直接接其参数。

另从文言的视觉效果来看，单字标识符还加上两引号，太浪费，而长标识符两边加引号则没那么影响观感。

就编程实践而论，在实现通用算法或（及）标准库时常需用到临时变量，可用单字标识符（对应主流语言的 i j k），如能略去引号，更有范本风格。在具体工程项目中，才推荐使用长变量名。
